### DISCUSSION TITLE
Question : Is there any way we could create different interfaces for different sql files??

### DISCUSSION THREAD

**Discussion by sagar23sj** [sqlc-dev/sqlc#997] [Q&A]
Current Behaviour : 
I have 3 .sql files and when I am compiling them using sqlc generate(keeping emit_interface flag true), a single interface is created for all the auto-generated SQL queries.

Expected Behaviour :
Wanted to have different interfaces for different table operations.

Not having separate interface for each SQL file is creating a common interface which gets bulky everytime I add a query.

Can anyone suggest any way if possible to generate separate interfaces for separate SQL files?

Thanks

**Comment by Kangaroux** [comment]
Having a separate `Querier` interface for each file would make test mocking easier. `Querier` on its own isn't terribly useful for APIs that are treating each table as a resource. I think the only suitable workaround is to create your own interface for each table/resource/etc. which is a subset of `Querier`.
